File: bindings/csharp/Sop/Relation.cs
using System.Collections.Generic;
using System.Text.Json.Serialization;

namespace Sop;

/// <summary>
/// Describes a foreign key relationship to another store.
/// </summary>
public class Relation
{
    /// <summary>
    /// The fields in the current store that act as the foreign key.
    /// </summary>
    [JsonPropertyName("source_fields")]
    public List<string> SourceFields { get; set; }

    /// <summary>
    /// The name of the target store usually "parent" table.
    /// </summary>
    [JsonPropertyName("target_store")]
    public string TargetStore { get; set; }

    /// <summary>
    /// The fields in the target store that match the source fields (usually the primary key).
    /// </summary>
    [JsonPropertyName("target_fields")]
    public List<string> TargetFields { get; set; }

    public Relation() { }

    public Relation(List<string> sourceFields, string targetStore, List<string> targetFields)
    {
        SourceFields = sourceFields;
        TargetStore = targetStore;
        TargetFields = targetFields;
    }
}

File: bindings/java/src/main/java/com/sharedcode/sop/Relation.java
package com.sharedcode.sop;

import java.util.List;
import com.fasterxml.jackson.annotation.JsonProperty;

public class Relation {
    @JsonProperty("source_fields")
    public List<String> sourceFields;

    @JsonProperty("target_store")
    public String targetStore;

    @JsonProperty("target_fields")
    public List<String> targetFields;

    public Relation() {
    }

    public Relation(List<String> sourceFields, String targetStore, List<String> targetFields) {
        this.sourceFields = sourceFields;
        this.targetStore = targetStore;
        this.targetFields = targetFields;
    }
}
